### PR TITLE
DLT runtimes display fix and issue #182

### PR DESCRIPTION
**Pull request by souvik-databricks**
This pull request contains the bug fix for the display function for DLT runtimes and the change for github issue #182

### CODE REVIEW

File: python/tempo/__init__.py [hunk 1]
```diff
@@ -1,2 +1,14 @@
+from IPython import get_ipython
+
 from tempo.tsdf import TSDF
-from tempo.utils import display
\ No newline at end of file
+if (
+            ('create_dlt_table_fn' not in list(get_ipython().user_ns.keys()))
+            and
+            ('dlt_sql_fn' not in list(get_ipython().user_ns.keys()))
```
Comment (tnixon) [inline python/tempo/__init__.py:7]: You shouldn't need to construct this list twice. How about:

```
user_ns_labels = list(get_ipython().user_ns.keys())

if ( ('create_dlt_table_fn' not in user_ns_labels) and ('dlt_sql_fn' not in user_ns_labels))
[...]
```

File: python/tempo/utils.py [hunk 2]
```diff
@@ -57,19 +57,29 @@ def display_unavailable(df):
...
-
-
-    display = display_improvised
+    # This below check is for ensuring compatibility with Databricks DLT runtimes
+    # This if logic ensures that the custom user's namespace of DLT runtimes
+    # which doesn't have PythonShell's Display object in the namespace doesn't result in an error.
+    if (
+            ('create_dlt_table_fn' not in list(get_ipython().user_ns.keys()))
+            and
+            ('dlt_sql_fn' not in list(get_ipython().user_ns.keys()))
```
Comment (tnixon) [inline python/tempo/utils.py:67]: You're performing this check multiple times, you should use a function.
Maybe create a function in `__init__.py` called something like `__is_display_fn_avail()` where you have this check.

Simple rule to follow with your code: any time you find yourself writing/copying the same code - it's time for a function (or global var)

**Review by tnixon — CHANGES_REQUESTED** [review]
A couple of small changes needed, please see the inline comments.

File: python/tempo/__init__.py (hunk 1 above)
Comment (souvik-databricks) [reply python/tempo/__init__.py:7]: I agree I will make the change @tnixon

File: python/tempo/utils.py (hunk 2 above)
Comment (souvik-databricks) [reply python/tempo/utils.py:67]: Yes makes sense I will move this to a function

File: python/tempo/__init__.py (hunk 1 above)
Comment (souvik-databricks) [reply python/tempo/__init__.py]: pushed the changes

File: python/tempo/utils.py (hunk 2 above)
Comment (souvik-databricks) [reply python/tempo/utils.py]: made this into a function

File: python/tempo/utils.py [hunk 3]
```diff
@@ -17,6 +17,12 @@
 """
 
 
+def __notdltruntime():
```
[the remark this answers is not in the document]
Comment (souvik-databricks) [reply python/tempo/utils.py]: thanks for the suggestions and the help @tnixon. Now the code much cleaner and more streamlined.

**Review by rportilla-databricks — APPROVED** [review]

**Review by tnixon — APPROVED** [review]
Looks good!

**Pull request merged**